ISISScientificComputing/autoreduce
# Add unit tests to webapp

### Summary of work
- Fixes up Pytest to work gracefully with Django.
- Adds an example test
- Does some fancy rebase and merge to get the commits from #657 into `master` to avoid a merge conflict when develop goes in too   (note this this PR was created develop branch has been deleted)


Now developers like myself don't have an excuse for not writing unit tests for the web app :wink: 

### How to test your work
- Install the new Python packages requires from `requirements.txt`
- Locally, try running `pytest WebApp` (or alternative `pytest` in your root dir if testing is setup for other frameworks). This should "just work"

### Additional comments
- TODO Re-run all our migrations each time we test to ensure they actually work, rather than hoping we can ever get back the same schema

**Before merging ensure the release notes have been updated**
Does this apply to non-user facing changes? ^
